pkg/cli/nodes_connectivity.go

```diff
@@ -0,0 +1,403 @@
+	id := uuid.New().String()
+	options := []plumber.Option{
+		plumber.WithKustomizeMutator(kustomizeMutator(opts)),
+		plumber.WithObjectMutator(func(ctx context.Context, obj client.Object) error {
+			if job, ok := obj.(*batchv1.Job); ok {
+				env := []corev1.EnvVar{
+					{Name: "NODEIP", Value: targetIP},
+					{Name: "NODEPORT", Value: fmt.Sprint(opts.port)},
+					{Name: "UUID", Value: id},
+				}
+				args := corev1.EnvVar{Name: "NCARGS", Value: "-w 2"}
+				if corev1.Protocol(opts.proto) == corev1.ProtocolUDP {
+					args = corev1.EnvVar{Name: "NCARGS", Value: "-uw 2"}
+				}
+				env = append(env, args)
+				job.Spec.Template.Spec.Affinity = model.Spec.Affinity
+				//job.Spec.Template.Spec.Tolerations = model.Spec.Tolerations
+				job.Spec.Template.Spec.Containers[0].Env = env
+			}
+			return nil
+		}),
+		plumber.WithPostApplyAction(func(ctx context.Context, obj client.Object) error {
+			if job, ok := obj.(*batchv1.Job); ok {
+				if _, err := k8sutil.WaitForJob(ctx, opts.cliset, job, time.Minute); err != nil {
+					return fmt.Errorf("failed to create job: %s", err)
+				}
+			}
+			return nil
+		}),
+	}
+	renderer := plumber.NewRenderer(opts.cli, nodes_connectivity.Static, options...)
+	if err := renderer.Apply(ctx, "pinger"); err != nil {
+		return "", fmt.Errorf("failed to apply pinger job: %w", err)
+	}
+	return id, nil
+}
+
+// testNodesConnectivity tests the connectivity between the cluster nodes.
+func testNodesConnectivity(ctx context.Context, opts nodeConnectivityOptions) error {
+	pods, err := k8sutil.ListPodsBySelector(ctx, opts.cliset, opts.namespace, listenersSelector)
+	if err != nil {
+		return fmt.Errorf("failed to get listener pods: %w", err)
+	}
+	receiver := attachToListenersPods(ctx, opts, pods.Items)
+	var nodes corev1.NodeList
+	if err := opts.cli.List(ctx, &nodes); err != nil {
+		return fmt.Errorf("failed to list nodes: %w", err)
+	}
+	for _, node := range nodes.Items {
+		if err := connectToNodeFromPods(ctx, opts, node, receiver); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// connectToNodeFromPods connects to the provided node by spawning pods in all other nodes and verifying they can reach
+// the destination ip address and port.
+func connectToNodeFromPods(ctx context.Context, opts nodeConnectivityOptions, node corev1.Node, receiver <-chan logLine) error {
+	pods, err := k8sutil.ListPodsBySelector(ctx, opts.cliset, opts.namespace, listenersSelector)
+	if err != nil {
+		return fmt.Errorf("failed to get listener pods: %w", err)
+	}
+	dstIP, err := k8sutil.NodeInternalIP(node)
+	if err != nil {
+		return fmt.Errorf("failed to determine node %s ip: %w", node.Name, err)
+	}
+	attempts := opts.attempts
+	if corev1.Protocol(opts.proto) == corev1.ProtocolTCP {
+		attempts = 1
+	}
+	for _, pod := range pods.Items {
+		if pod.Spec.NodeName == node.Name {
+			continue
+		}
+		var success bool
+		src, dst := pod.Spec.NodeName, node.Name
```

Comment:
this feels a little more explicit

```suggestion
		src, dst := pod.Spec.NodeName, node.Name
		if src == dst {
			continue
		}
		var success bool
```

Reply:
Done.